core/src/main/java/io/questdb/griffin/engine/table/AsyncFilterAtom.java

```diff
@@ -117,4 +118,73 @@ public void releaseFilter(int filterId) {
         }
         perWorkerLocks.set(filterId, 0);
     }
+
+    /**
+     * Pre-touches fixed-size columns, if the feature is configured.
+     * <p>
+     * The idea is to access the memory to page fault and, thus, warm up the pages
+     * in parallel, on multiple threads, instead of relying on the "query owner" thread
+     * to do it later serially.
+     */
+    public void preTouchColumns(PageAddressCacheRecord record, DirectLongList rows) {
+        if (preTouchColumnTypes == null) {
+            return;
+        }
+        // We use a LongAdder as a blackhole to make sure that the JVM JIT compiler keeps the load instructions in place.
+        long sum = 0;
+        for (long p = 0; p < rows.size(); p++) {
+            long r = rows.get(p);
+            record.setRowIndex(r);
+            for (int i = 0; i < preTouchColumnTypes.size(); i++) {
+                int columnType = preTouchColumnTypes.getQuick(i);
+                switch (ColumnType.tagOf(columnType)) {
+                    case ColumnType.BOOLEAN:
+                        sum += record.getBool(i) ? 1 : 0;
+                        break;
+                    case ColumnType.BYTE:
+                        sum += record.getByte(i);
+                        break;
+                    case ColumnType.SHORT:
+                        sum += record.getShort(i);
+                        break;
+                    case ColumnType.INT:
+                    case ColumnType.SYMBOL: // We're interested in pre-touching the page only, so we read symbol key only.
+                        sum += record.getInt(i);
+                        break;
+                    case ColumnType.LONG:
+                    case ColumnType.DATE:
+                    case ColumnType.TIMESTAMP:
+                        sum += record.getLong(i);
+                        break;
+                    case ColumnType.FLOAT:
+                        sum += record.getFloat(i);
+                        break;
+                    case ColumnType.DOUBLE:
+                        sum += record.getDouble(i);
+                        break;
+                    case ColumnType.LONG256:
```

Comment:
we don't pre-touch `string` columns, perhaps we should?

Reply:
Pre-touching string and binary types would be more costly in case if if the data is in the page cache. On the other hand, there would be no memory copying required since we use views over the mmapped memory. Do you think I should pre-touch these types too?

Reply:
I think we should, some users rely on String columns

Reply:
Done in f5efd8d0fa638ff1b4c1dabb04bc12d8ee74b2e5